src/inspector_agent.cc

```diff
@@ -437,18 +418,40 @@ void Agent::OnRemoteDataIO(uv_stream_t* stream,
   uv_cond_broadcast(&agent->pause_cond_);
 }
 
+void Agent::PushPendingMessage(std::vector<std::string>* queue,
+                               const std::string& message) {
+  uv_mutex_lock(&queue_lock_);
+  queue->push_back(message);
+  uv_mutex_unlock(&queue_lock_);
+}
+
+std::vector<std::string> Agent::TakePendingMessages(
+        std::vector<std::string>* queue) {
```

Comment:
I also decided to take a member pointer, to signify that the lock is for the object state.